### PR TITLE
Add client generation for one:many relations

### PR DESCRIPTION
**Pull request by sahanHe**
## Purpose
Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/3587

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_17/main.bal [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import ballerina/io;
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_17/main.bal:1]: add license headers

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_16/main.bal [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import ballerina/io;
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_16/main.bal:1]: Add license headers

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalFileConstants.java [hunk 3]
```diff
@@ -35,11 +35,20 @@ private BalFileConstants() {
...
     public static final String SPACE_STRING = " ";
     public static final String COMMA_SPACE_STRING = ", ";
+    public static final String FIELD_ACCESSS_STRING = ".";
+    public static final String ARRAY_TYPE_STRING = "[]";
+    public static final String START_CLOSED_RECORD_STRING = "{|";
+    public static final String END_CLOSED_RECORD_STRING = "|}";
+    public static final String START_RECORD_STRING = "{";
+    public static final String END_RECORD_STRING = "}";
+    public static final String COLON_STRING = ":";
+    public static final String SEMI_COLON_STRING = ";";
```
Comment (MadhukaHarith92) [inline persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalFileConstants.java:49]:
```suggestion
    public static final String SEMICOLON_STRING = ";";
```

Comment (MadhukaHarith92) [inline persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalFileConstants.java:49]: Do we need "STRING" as a suffix?

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_16/main.bal (hunk 2 above)
Comment (sahanHe) [reply persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_16/main.bal:17]: Updated in [704a43f](https://github.com/ballerina-platform/persist-tools/pull/46/commits/704a43f3344de31ca790267f6ba9cc31ad09d26f)

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalFileConstants.java (hunk 3 above)
Comment (sahanHe) [reply persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalFileConstants.java]: Updated in [704a43f](https://github.com/ballerina-platform/persist-tools/pull/46/commits/704a43f3344de31ca790267f6ba9cc31ad09d26f)

Comment (sahanHe) [reply persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalFileConstants.java]: It's not required. Removed in [704a43f](https://github.com/ballerina-platform/persist-tools/pull/46/commits/704a43f3344de31ca790267f6ba9cc31ad09d26f)

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_17/main.bal (hunk 1 above)
Comment (daneshk) [reply persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_17/main.bal:17]: Shall we use LLC format

Comment (sahanHe) [reply persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_17/main.bal:17]: updated in [682be05](https://github.com/ballerina-platform/persist-tools/pull/46/commits/682be05657faaa87c11fcd4c856559e1b84b2367)

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_16/modules/clients/entities.bal [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import ballerina/persist;
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_16/modules/clients/entities.bal:1]: Add licenses?

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java [hunk 5]
```diff
@@ -191,34 +222,57 @@ public static EntityMetaData getEntityRecord(Path filePath, Optional<String> mod
...
+                    }
                     FieldMetaData field;
                     if (((RecordFieldWithDefaultValueNode) node).metadata().isEmpty()) {
                         field = new FieldMetaData(fName, fType, false);
                     } else {
                         MetadataNode fieldMetaD = ((RecordFieldWithDefaultValueNode) node).metadata().get();
                         Relation relation = readMetaData(fName, fType, fieldMetaD);
                         field = new FieldMetaData(fName, fType, checkAutoIncrement(fieldMetaD));
-                        if (!relation.getKeyColumns().isEmpty() && !relation.getReferences().isEmpty()) {
+                        if (!(relation.getKeyColumns() == null) && !(relation.getReferences() == null)) {
```
Comment (daneshk) [inline persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java:241]:
```suggestion
                        if (relation.getKeyColumns() != null && relation.getReferences() != null) {
```

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java [hunk 6]
```diff
@@ -191,34 +222,57 @@ public static EntityMetaData getEntityRecord(Path filePath, Optional<String> mod
...
+                    }
                     FieldMetaData field;
                     if (((RecordFieldNode) node).metadata().isEmpty()) {
                         field = new FieldMetaData(fName, fType, false);
                     } else {
                         MetadataNode fieldMetaD = ((RecordFieldNode) node).metadata().get();
                         Relation relation = readMetaData(fName, fType, fieldMetaD);
                         field = new FieldMetaData(fName, fType, checkAutoIncrement(fieldMetaD));
-                        if (!relation.getReferences().isEmpty() || !relation.getKeyColumns().isEmpty()) {
+                        if (!(relation.getKeyColumns() == null) && !(relation.getReferences() == null)) {
```
Comment (daneshk) [inline persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java:268]:
```suggestion
                        if (relation.getKeyColumns() != null && relation.getReferences() != null) {
```

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java [hunk 7]
```diff
@@ -993,9 +1083,25 @@ public static void generateRelations(ArrayList<Entity> entityArray) {
                             boolean childIncludesParentColumns = false;
                             ArrayList<Integer> indexesToRemove = new ArrayList<>();
                             for (FieldMetaData fieldMetaData : entity.getFields()) {
-                                if (fieldMetaData.getFieldType().contains(":")) {
-                                    if (entityNames.contains(fieldMetaData.getFieldType().split(":", 2)[1])) {
-                                        fieldMetaData.setFieldType(fieldMetaData.getFieldType().split(":", 2)[1]);
+                                if (fieldMetaData.getFieldType().contains(COLON)) {
+                                    if (entityNames.contains(fieldMetaData.getFieldType().split(COLON, 2)[1])) {
+                                        fieldMetaData.setFieldType(fieldMetaData.getFieldType().split(COLON,
+                                                2)[1]);
```
Comment (daneshk) [inline persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java:1089]:
```suggestion
                                        fieldMetaData.setFieldType(fieldMetaData.getFieldType().split(COLON, 2)[1]);
```

**Review by daneshk — APPROVED** [review]

**Pull request merged**